core/src/main/java/org/apache/accumulo/core/clientImpl/ScannerImpl.java

```diff
@@ -163,7 +163,7 @@ public synchronized int getBatchSize() {
   public synchronized Iterator<Entry<Key,Value>> iterator() {
     ensureOpen();
     ScannerIterator iter = new ScannerIterator(context, tableId, authorizations, range, size,
-        getTimeout(SECONDS), this, isolated, readaheadThreshold, new Reporter());
+        Duration.ofMillis(retryTimeout), this, isolated, readaheadThreshold, new Reporter());
```

Comment:
With the following change less code is making assumptions about the units of the `retryTimeout` instance variable.

```suggestion
        Duration.ofMillis(getTimeout(MILLISECONDS)), this, isolated, readaheadThreshold, new Reporter());
```

I was wondering if `getTimeout()` could be converted to return a Duration. Looking at how its used that kinda cascaded, which is what I think you mentioned in the PR description.

Reply:
Good point on using getTimeout() method, I'll change that.

In regards to converting getTimeout() to a Duration...right that is the exact problem I mentioned in my PR, it cascades very quickly and starts to impact several other classes as the timeouts are passed around so changing it to a Duration means changing a lot of other stuff. Also, for example, there's also getBatchTimeout() and I don't think it makes sense to change one and not the other.

So that's why I was thinking it would be better to minimize the changes in 2.1 at this point. We could make it a follow on for 3.1 and main. However, if it you think it's worthwhile for 2.1 I can still make the changes in 2.1 (even if it's another PR)